Documentation/project-docs/developer-guide.md

```diff
@@ -3,6 +3,12 @@ Developer Guide
 
 This guide provides instructions (mostly as links) on how to build the repo and implement improvements. It will expand over time.
 
+Dev Workflow
+===============
```

Comment:
(nit) I think using a slightly smaller font would look a bit better for both the "Dev Workflow" and "Building the repository" headings.  This would be more consistent with our other documentation, e.g. [contributing.md](https://github.com/dotnet/corefx/blob/master/Documentation/project-docs/contributing.md). This is specified by using `-` characters instead of `=` characters.